### PR TITLE
Add Support for custom entrypoint

### PR DESCRIPTION
**Pull request by soloman1124**
Hi, I think in some cases it is useful to run the whalebrew package with custom entrypoint. For example, instead of running the target command directly, sometimes it is necessary to access shell and execute the target command within the docker shell. This PR add the following support:

```
whalebrew run --entrypoint=/bin/sh $(which jq)
```

### CODE REVIEW

**Comment by shnhrrsn** [comment]
I’d actually love to see both of these, file would be ideal for images without an entrypoint and doing it at runtime to quickly override would be nice as well.

**Comment by soloman1124** [comment]
hi, @bfirsh one particular use cause I am currently doing is that I have a `yarn` whalebrew package (which is based on standard node image). It works 95% of time, however, occasionally I want to do other commands such as `npm publish`, it would be handy to use same image for that command. 

Similarly there are some applications with multiple command line tools, but most of them are rarely used, in these cases, having shell access is very handy.

**Comment by soloman1124** [comment]
@bfirsh and @shnhrrsn I've also added the support for additional `entrypoint` configuration in the file, as what you've suggested. 

Just wondering if you'd merge it~

**Comment by bfirsh** [comment]
Great, thanks!

Okay, this sounds good in principle. I'm not sure about parsing CLI options with a regexp though. Is there a reason this can't be done with a parsing library?

**Comment by bfirsh** [comment]
Travis build is breaking for some reason too, hrm. Might be unrelated?

**Comment by soloman1124** [comment]
@bfirsh thanks for the suggestion. I will take  another look. 

Btw, I think the breaking build is unrelated.

**Comment by soloman1124** [comment]
@bfirsh I've removed the regex parsing and stick with CLI option parsing as you suggested. Please take another review. 

Again, the failing travis-ci seems not related to my PR.

**Comment by soloman1124** [comment]
@bfirsh I've fixed the travis build by updating the golang version. let me know if there are any other suggestions for this PR. Thanks!

**Comment by bfirsh** [comment]
Ah, I misunderstood this at first glance.

`whalebrew run` is not designed to be used by a user, and I think it breaks the abstraction of the tool if we start adding features to it. Having to document `whalebrew run` would get very confusing, because now there are two entrypoints to running commands.

Perhaps we could use environment variables for this: `WHALEBREW_ENTRYPOINT=/bin/sh jq`. It is a weird user interface. Perhaps it's okay if we just consider it a power user feature.

So in summary: +0 for the feature, -1 for design. Sorry for the incorrect initial review - I've been very busy. :)

@tjamet would be curious what you think.

**Comment by bfirsh** [comment]
I cherry-picked your Travis fix to master though - thank you!

**Comment by tjamet** [comment]
My take is that we are having there 2 features:
1. being able to package multiple commands from a single image (the entrypoint in the yaml)
2. being able to debug the command line
I pretty buy the first one, as I encountered the exact same need with yarn/npm/node.
for the second one, I am unsure I would neither like to expose the `run` command

I am thinking that if we allow changing at runtime the entrypoint of a package, then this is another package, like a `jq-debug` one.

This said, we could have the first feature about customizing the entrypoint at install time. This allows debugging by copying/editing the executable, and later on have a `copy` function or similar that could serve to clone a package by changing some of its properties, including the entrypoint. This feature would also eventually be useful in the context of #17 to manage versions

**Comment by tjamet** [comment]
Hi, thanks for the suggestion, I cherry-picked the entry point override at the install time in #81 
For the runtime selector, I would prefer suggesting to edit the file or create a copy of it in order to enter the container and debug.
I am closing this PR as the ability to use a custom entry point has been added.
Thank you for your contribution
